### PR TITLE
docs: update docusaurus, add metrics endpoint, add zero downtime update

### PR DESCRIPTION
**Pull request by fforootd**

### CODE REVIEW

File: docs/docs/apis/observability/metrics.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+title: Metrics
```
Comment (mffap) [inline docs/docs/apis/observability/metrics.md:2]: @fforootd is there nothing available on ZITADEL Cloud?
Would it be useful to mention the HealthZ endpoints on the APIs?

File: docs/docs/apis/observability/metrics.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+---
+
+ZITADEL provides a `metrics` endpoint with the help of the [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go) package.
```
Comment (mffap) [inline docs/docs/apis/observability/metrics.md:5]: Does this require authentication?

File: docs/docs/concepts/architecture/solution.md [hunk 3]
```diff
@@ -57,3 +57,25 @@ CockroachDB needs to be configured with locality flags to proper distribute data
 
 ![Multi-Cluster Architecture](/img/zitadel_multicluster_architecture.png)
 
+## Zero Downtime Updates
+
+Since an Identity system tends to be a critical piece of infrastructure, the "in place zero downtime update" is a well needed feature.
+ZITADEL is built in a way that version can be updated without downtime by just updating to a more recent version.
```
Comment (mffap) [inline docs/docs/concepts/architecture/solution.md:63]:
```suggestion
ZITADEL is built in a way that upgrades can be executed without downtime by just updating to a more recent version.
```

File: docs/docs/concepts/architecture/solution.md [hunk 4]
```diff
@@ -57,3 +57,25 @@ CockroachDB needs to be configured with locality flags to proper distribute data
...
+
+The common update involves the following steps and do not need manual intervention of the operator:
+
+- Keep the old version running
+- Deploy the version parallel to the old version
```
Comment (mffap) [inline docs/docs/concepts/architecture/solution.md:68]:
```suggestion
- Deploy the version in parallel to the old version
```

File: docs/docs/concepts/architecture/solution.md [hunk 5]
```diff
@@ -57,3 +57,25 @@ CockroachDB needs to be configured with locality flags to proper distribute data
...
+- The new version will start ...
+  - by updating databases schemas if needed
+  - participate in the leader election for background jobs
+- As soon as the new version is ready to accept traffic it will signal this on the readiness endpoint `/debug/ready` 
+- At this point your network infrastructure can send traffic to the new version
+
+Customer who use [Kubernetes/Helm](../../guides/deploy/kubernetes) or serverless container services like Google Cloud Run can benefit from the fact the above process is automated.
```
Comment (mffap) [inline docs/docs/concepts/architecture/solution.md:75]:
```suggestion
Users who use [Kubernetes/Helm](../../guides/deploy/kubernetes) or serverless container services like Google Cloud Run can benefit from the fact the above process is automated.
```

File: docs/docs/apis/observability/metrics.md (hunk 1 above)
Comment (fforootd) [reply docs/docs/apis/observability/metrics.md:2]: Yes, the cloud does not expose the endpoint at the moment, but we could make exceptions if needed.

Healthz and Readiness I can append something.

File: docs/docs/apis/observability/metrics.md (hunk 2 above)
Comment (fforootd) [reply docs/docs/apis/observability/metrics.md:5]: No, not at the moment.

File: docs/docs/apis/observability/metrics.md (hunk 1 above)
Comment (mffap) [reply docs/docs/apis/observability/metrics.md:2]: That would be great! Just as a pointer / hint.

Comment (fforootd) [reply docs/docs/apis/observability/metrics.md:2]: Its done!

Comment (fforootd) [reply docs/docs/apis/observability/metrics.md:2]: https://github.com/zitadel/zitadel/pull/4376/files#diff-b7564c95acfd8781351c5ae69d9dd403deb0ac925282ab0e50b2e8c307c47abc

File: docs/docs/apis/observability/health.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+title: Ready / Healthy
+---
+
+ZITADEL exposes a `Ready`- and `Health` endpoint to allow external systems like load balancers, orchestration systems, uptime probes and others to check the status.
```
Comment (mffap) [inline docs/docs/apis/observability/health.md:5]:
```suggestion
ZITADEL exposes a `Ready`- and `Healthy` endpoint to allow external systems like load balancers, orchestration systems, uptime probes and others to check the status.
```

**Review by mffap — DISMISSED** [review]
@fforootd one small typo, but rest looks good.

**Pull request merged**